### PR TITLE
[test]Add test cases to rclnodejs

### PR DESCRIPTION
**Pull request by qiuzhong**
Add 18 test cases from rclpy to rclnodejs.

Signed-off-by: Qiu Zhong <zhongx.qiu@intel.com>

### CODE REVIEW

File: test/test-destruction.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        return rclnodejs.init();
+    });
+
+    after(function() {
+        rclnodejs.shutdown();
+    });
+
+    it('node.destroy()', function() {
+        var node = rclnodejs.createNode('my_node1');
+        node.destory();
```
Comment (Honry) [inline test/test-destruction.js]: Please use the correct spelling.

File: test/test-destruction.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    });
+
+    it('node.destroy() corrupted', function() {
+        var node = rclnodejs.createNode('my_node3');
+        node.destory();
+
+        var org_handle = node._handle;
+        node._handle = 'garbage'; 
+        try {               
+                node.destory();
```
Comment (Honry) [inline test/test-destruction.js]: Indent error.

Comment (Honry) [reply test/test-destruction.js]: Please use two whitespace indent as common convention in JavaScript.

File: test/test-node.js [hunk 3]
```diff
@@ -27,8 +115,89 @@ describe('rclnodejs node test suite', function() {
...
+    }, /unknown substitution/, 'Failed to createClient');
+  });
+
+  it('node.createService', function() {
+    var GetParameters = rclnodejs.require('rcl_interfaces').srv.GetParameters;
+    node.createService(GetParameters, 'get/parameters', () => {
+      assert.ok(true);
+    });
+
+    assert.throws(function() {
```
Comment (Honry) [inline test/test-node.js]: Move invalid names to an array and use `foreach` to invoke assertion to make test robust.

**Comment by minggangw** [comment]
Please add `mocha` as `devDependencies` in package.json to restrain the ci failure.

File: test/test-destruction.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+// limitations under the License.
+
+'use strict';
+
+const assert = require('assert');
+const rclnodejs = require('../index.js');
+
+describe('Node destroy testing', function() {
+  before(function() {
+    this.timeout(10 * 1000);
```
Comment (minggangw) [inline test/test-destruction.js]: As the performance of the machine used by travis-ci seems not as good as our dev one, the 10s time out is not enough to generate all the necessary js files. I suggest set the time out value to 60s as what I have done on the development branch.

File: test/test-init-shutdown.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+// limitations under the License.
+
+'use strict';
+
+const assert = require('assert');
+const rclnodejs = require('../index.js');
+
+describe('Node destroy testing', function() {
+  before(function() {
+    this.timeout(10 * 1000);
```
Comment (minggangw) [inline test/test-init-shutdown.js]: ditto

File: test/test-node.js [hunk 6]
```diff
@@ -18,6 +18,90 @@ const assert = require('assert');
 const rclnodejs = require('../index.js');
 
 describe('rclnodejs node test suite', function() {
+  describe('createNode method testing', function() {
+    before(function() {
+      this.timeout(10 * 1000);
```
Comment (minggangw) [inline test/test-node.js]: ditto

File: test/test-node.js [hunk 7]
```diff
@@ -18,6 +18,90 @@ const assert = require('assert');
...
+    });
+  });
+});
+
+describe('rcl node methods testing', function() {
+  var node;
+  var rclString;
+
   before(function() {
     this.timeout(10 * 1000);
```
Comment (minggangw) [inline test/test-node.js]: ditto

File: test/test-init-shutdown.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+      assert.ok(false);
+      done(err);
+    });        
+  });
+
+  it('rclnodejs double init', function(done) {
+    rclnodejs.init().then(function() {
+      assert.ok(true);    
+    }).then(function() {
+      rclnodejs.init();
```
Comment (minggangw) [inline test/test-init-shutdown.js]: As the `init` method returns a Promise, you have to catch the error here OR return it to keep all the promises in the same chain.

File: test/test-init-shutdown.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+      assert.ok(false);
+      done();
+    }).catch(function(err) {
+      assert.ok(true);
+      rclnodejs.shutdown();
+      done(err);
```
Comment (minggangw) [inline test/test-init-shutdown.js]: Pass the `err` will fail the test although the result is expected.

**Review by minggangw — COMMENTED** [review]
After PR#69 is merged, and the problems here are fixed, all the unit tests can run correctly as I have run them locally. Would you please merge these three patches on this branch together when updating? thanks!

File: test/test-init-shutdown.js [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    });
+  });
+
+  it('rclnodejs create node without init', function() {
+    assert.throws(function() {
+      rclnodejs.createNode('my_node');
+    }, /has not been called/, 'Failed to createNode');
+  });
+});
```
Comment (minggangw) [inline test/test-init-shutdown.js:98]: invalid ending

File: test/test-destruction.js (hunk 4 above)
Comment (qiuzhong) [reply test/test-destruction.js]: Sure. I'll update it.

File: test/test-init-shutdown.js (hunk 10 above)
Comment (qiuzhong) [reply test/test-init-shutdown.js:98]: Do you mean the `shutdown()` method is not called for cleanup?

File: test/test-init-shutdown.js (hunk 9 above)
Comment (qiuzhong) [reply test/test-init-shutdown.js]: Yes. I updated but it still got failed. Do you have any good suggestion?

File: test/test-init-shutdown.js (hunk 8 above)
Comment (qiuzhong) [reply test/test-init-shutdown.js]: Thanks! I'll update it.

File: test/test-init-shutdown.js (hunk 9 above)
Comment (qiuzhong) [reply test/test-init-shutdown.js]: Got it. I should not pass `err` the `done()`.

**Review by minggangw — APPROVED** [review]
LGTM, thanks for the patch which makes the model more robust!

**Pull request merged**